Logger object null in DefaultPrincipalAttributesRepository class

HI,

While deploying cas 4.2.1 with jpaServiceRegistryDao we got an exception during the "Attribute release" process ( after authentication)
No error occurs using jsonServiceRegistryDao.

The logger object used in class DefaultPrincipalAttributesRepository is null.

CAS 4.2.1
Tomcat 8

Stack trace :
`=============================================================
WHO: test
WHAT: Supplied credentials: [test]
ACTION: AUTHENTICATION_SUCCESS
APPLICATION: CAS
WHEN: Tue May 03 15:44:01 CEST 2016
CLIENT IP ADDRESS: 134.206.1.214
SERVER IP ADDRESS: test-cas-ng.univ-lille1.fr
=============================================================

>
2016-05-03 15:44:01,756 DEBUG [org.jasig.cas.authentication.DefaultAuthenticationTransactionManager] - <Successful authentication; Collecting authentication result [org.jasig.cas.authentication.ImmutableAuthentication@6599186d]>
2016-05-03 15:44:01,757 DEBUG [org.jasig.cas.authentication.DefaultAuthenticationTransactionManager] - <Transaction ignored since there are no credentials to authenticate>
2016-05-03 15:44:01,757 DEBUG [org.jasig.cas.authentication.DefaultAuthenticationContextBuilder] - <Collecting authentication history based on [1] authentication events>
2016-05-03 15:44:01,757 DEBUG [org.jasig.cas.authentication.DefaultAuthenticationContextBuilder] - <Evaluating authentication principal [test] for inclusion in context>
2016-05-03 15:44:01,758 DEBUG [org.jasig.cas.authentication.DefaultAuthenticationContextBuilder] - <Collected principal attributes [{mail=mail, sn=sn, cn=name, supannEtuId=nip, givenname=givenname, displayName=displayName, supannEmpId=matricule}] for inclusion in context for principal [test]>
2016-05-03 15:44:01,758 DEBUG [org.jasig.cas.authentication.DefaultAuthenticationContextBuilder] - <Collected single authentication attribute [authenticationMethod] -> [LdapAuthenticationHandler]>
2016-05-03 15:44:01,758 DEBUG [org.jasig.cas.authentication.DefaultAuthenticationContextBuilder] - <Collected single authentication attribute [successfulAuthenticationHandlers] -> [[LdapAuthenticationHandler]]>
2016-05-03 15:44:01,758 DEBUG [org.jasig.cas.authentication.DefaultAuthenticationContextBuilder] - <Finalized authentication attributes [{authenticationMethod=LdapAuthenticationHandler, successfulAuthenticationHandlers=[LdapAuthenticationHandler]}] for inclusion in authentication context>
2016-05-03 15:44:01,759 DEBUG [org.jasig.cas.authentication.DefaultPrincipalElectionStrategy] - <Nominated [test] as the primary principal>
2016-05-03 15:44:01,759 DEBUG [org.jasig.cas.authentication.DefaultAuthenticationContextBuilder] - <Determined primary authentication principal to be [test]>
2016-05-03 15:44:01,759 DEBUG [org.jasig.cas.authentication.DefaultAuthenticationContextBuilder] - <Collected authentication attributes for this context are [{authenticationMethod=LdapAuthenticationHandler, successfulAuthenticationHandlers=[LdapAuthenticationHandler]}]>
2016-05-03 15:44:01,760 DEBUG [org.jasig.cas.authentication.DefaultAuthenticationContextBuilder] - <Authentication context commenced at [2016-05-03T15:44:01.759+02:00]>
2016-05-03 15:44:01,760 DEBUG [org.jasig.cas.authentication.DefaultAuthenticationContextBuilder] - <Building an authentication context for authentication org.jasig.cas.authentication.ImmutableAuthentication@6a0070bd and service https://idp-test.univ-lille1.fr/idp/Authn/ExtCas?conversation=e1s1>
2016-05-03 15:44:01,776 DEBUG [org.jasig.cas.ticket.registry.MemCacheTicketRegistry] - <Adding ticket TGT-1-TWE4GGCVtDhTwcPPrGGQcsqjxH5zyf5ryVR20w1U4MLYLESKpW-cas1-ng.univ-lille1.fr>
2016-05-03 15:44:01,831 DEBUG [org.jasig.cas.CentralAuthenticationServiceImpl] - <Publishing org.jasig.cas.support.events.CasTicketGrantingTicketCreatedEvent@79b5cf54[ticketGrantingTicket=TGT-1-TWE4GGCVtDhTwcPPrGGQcsqjxH5zyf5ryVR20w1U4MLYLESKpW-cas1-ng.univ-lille1.fr]>
2016-05-03 15:44:01,842 DEBUG [org.jasig.cas.audit.spi.TicketOrCredentialPrincipalResolver] - <Resolving argument [DefaultAuthenticationContext] for audit>
2016-05-03 15:44:01,844 INFO [org.jasig.inspektr.audit.support.Slf4jLoggingAuditTrailManager] - <Audit trail record BEGIN
=============================================================
WHO: audit:unknown
WHAT: TGT-1-TWE4GGCVtDhTwcPPrGGQcsqjxH5zyf5ryVR20w1U4MLYLESKpW-cas1-ng.univ-lille1.fr
ACTION: TICKET_GRANTING_TICKET_CREATED
APPLICATION: CAS
WHEN: Tue May 03 15:44:01 CEST 2016
CLIENT IP ADDRESS: 134.206.1.214
SERVER IP ADDRESS: test-cas-ng.univ-lille1.fr
=============================================================

>
2016-05-03 15:44:01,867 DEBUG [org.jasig.cas.web.support.WebUtils] - <Evaluating request to determine if warning cookie should be generated>
2016-05-03 15:44:01,868 DEBUG [org.jasig.cas.web.WarningCookieRetrievingCookieGenerator] - <Removed cookie with name [CASPRIVACY]>
2016-05-03 15:44:01,869 DEBUG [org.jasig.cas.web.flow.SendTicketGrantingTicketAction] - <Setting TGC for current session.>
2016-05-03 15:44:01,869 DEBUG [org.jasig.cas.web.support.DefaultCasCookieValueManager] - <Encoding cookie value [TGT-1-TWE4GGCVtDhTwcPPrGGQcsqjxH5zyf5ryVR20w1U4MLYLESKpW-cas1-ng.univ-lille1.fr@134.206.1.214@Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/50.0.2661.94 Safari/537.36]>
2016-05-03 15:44:01,870 DEBUG [org.jasig.cas.util.TGCCipherExecutor] - <Encrypting via [A128CBC-HS256]>
2016-05-03 15:44:01,874 DEBUG [org.jasig.cas.web.support.TGCCookieRetrievingCookieGenerator] - <Added cookie with name [TGC] and value [eyJhbGciOiJIUzUxMiJ9.WlhsS2FHSkhZMmxQYVVwcllWaEphVXhEU214aWJVMXBUMmxLUWsxVVNUUlJNRXBFVEZWb1ZFMXFWVEpKYmpBdUxuZFNlREI1ZWtwTGNIcE1hVkZGWm5sM1NFTkdWSGN1T0hoRFkzTTVTSFJzTUd0cWVtRjRMVU5VYUU5RE0yRm9hMlUwUkRGVmRtVnlOR3RKTUhWTk5VVTNTVnBKYUVob1ZreFBUV2czUlVSM1EycFVWbVZPVTB0VFZsSmxkblJJZUhGcFFURllWelpGT1hKSWNYRXdXbmsxVkMwNE1WZGliR1pJVEdGNmVHOXBPV3RoYmpoT01taFhWVU51YnpBeGIwVlhiMHRvUnpCR2FHdEVlbkpoVkRJM2NqWnliRzl3U1RSck5IY3RNV1paTVZGSFNUQnVlbEpxY2xsdGFXWjVXa05aYUMxNVNGUmxaV1ZrYWxwdmJqRkRZVGx0T0Voa2VUTXpaRUZTZDFkNk0wTlhSemRzZG5KSGRGWjJVbkExYkhOT1lVaExkMlIxUTBOTmVWbDRZMjV3UVZFMmRtbENWbXRsWXpGM0xXODNiMk5KTjJWUFpUTkRNRVJ3VFRJM1ptbEpiWGh6VUZoNE1scDNVR2N1ZEVWdU9YRkZWVTFGVFhvNGFrTlJZbUZoUzNBdGR3PT0.QHdxjwFmvzz9Q2rUhfv4wPpD6uZr0UBPx_-Zbj1NgmmkTjDs3da8Z7zPDzlxyywGKIl8qyj6MR0lP8prdHllyw]>
2016-05-03 15:44:01,919 DEBUG [org.jasig.cas.authentication.DefaultAuthenticationContextBuilder] - <Collecting authentication history based on [1] authentication events>
2016-05-03 15:44:01,920 DEBUG [org.jasig.cas.authentication.DefaultAuthenticationContextBuilder] - <Evaluating authentication principal [test] for inclusion in context>
2016-05-03 15:44:01,920 DEBUG [org.jasig.cas.authentication.DefaultAuthenticationContextBuilder] - <Collected principal attributes [{mail=mail, sn=sn, cn=name, supannEtuId=nip, givenname=givenname, displayName=displayName, supannEmpId=matricule}] for inclusion in context for principal [test]>
2016-05-03 15:44:01,920 DEBUG [org.jasig.cas.authentication.DefaultAuthenticationContextBuilder] - <Collected single authentication attribute [authenticationMethod] -> [LdapAuthenticationHandler]>
2016-05-03 15:44:01,920 DEBUG [org.jasig.cas.authentication.DefaultAuthenticationContextBuilder] - <Collected single authentication attribute [successfulAuthenticationHandlers] -> [[LdapAuthenticationHandler]]>
2016-05-03 15:44:01,920 DEBUG [org.jasig.cas.authentication.DefaultAuthenticationContextBuilder] - <Finalized authentication attributes [{authenticationMethod=LdapAuthenticationHandler, successfulAuthenticationHandlers=[LdapAuthenticationHandler]}] for inclusion in authentication context>
2016-05-03 15:44:01,921 DEBUG [org.jasig.cas.authentication.DefaultPrincipalElectionStrategy] - <Nominated [test] as the primary principal>
2016-05-03 15:44:01,921 DEBUG [org.jasig.cas.authentication.DefaultAuthenticationContextBuilder] - <Determined primary authentication principal to be [test]>
2016-05-03 15:44:01,921 DEBUG [org.jasig.cas.authentication.DefaultAuthenticationContextBuilder] - <Collected authentication attributes for this context are [{authenticationMethod=LdapAuthenticationHandler, successfulAuthenticationHandlers=[LdapAuthenticationHandler]}]>
2016-05-03 15:44:01,921 DEBUG [org.jasig.cas.authentication.DefaultAuthenticationContextBuilder] - <Authentication context commenced at [2016-05-03T15:44:01.921+02:00]>
2016-05-03 15:44:01,922 DEBUG [org.jasig.cas.authentication.DefaultAuthenticationContextBuilder] - <Building an authentication context for authentication org.jasig.cas.authentication.ImmutableAuthentication@9859bf07 and service https://idp-test.univ-lille1.fr/idp/Authn/ExtCas?conversation=e1s1>
2016-05-03 15:44:01,969 DEBUG [org.jasig.cas.audit.spi.TicketOrCredentialPrincipalResolver] - <Resolving argument [String] for audit>
2016-05-03 15:44:01,990 INFO [org.jasig.inspektr.audit.support.Slf4jLoggingAuditTrailManager] - <Audit trail record BEGIN
=============================================================
WHO: test
WHAT: https://idp-test.univ-lille1.fr/idp/Authn/ExtCas?conversation=e1s1
ACTION: SERVICE_TICKET_NOT_CREATED
APPLICATION: CAS
WHEN: Tue May 03 15:44:01 CEST 2016
CLIENT IP ADDRESS: 134.206.1.214
SERVER IP ADDRESS: test-cas-ng.univ-lille1.fr
=============================================================

>
2016-05-03 15:44:02,004 DEBUG [org.jasig.cas.web.FlowExecutionExceptionResolver] - <Ignoring the received exception due to a type mismatch>
org.springframework.webflow.execution.ActionExecutionException: Exception thrown executing org.jasig.cas.web.flow.GenerateServiceTicketAction@3ec2d801 in state 'generateServiceTicket' of flow 'login' -- action execution attributes were 'map[[empty]]'
        at org.springframework.webflow.execution.ActionExecutor.execute(ActionExecutor.java:60) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.action.EvaluateAction.doExecute(EvaluateAction.java:77) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.action.AbstractAction.execute(AbstractAction.java:188) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.execution.AnnotatedAction.execute(AnnotatedAction.java:145) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.execution.ActionExecutor.execute(ActionExecutor.java:51) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.ActionState.doEnter(ActionState.java:101) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.State.enter(State.java:194) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.Transition.execute(Transition.java:228) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.DecisionState.doEnter(DecisionState.java:51) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.State.enter(State.java:194) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.Transition.execute(Transition.java:228) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.impl.FlowExecutionImpl.execute(FlowExecutionImpl.java:395) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.impl.RequestControlContextImpl.execute(RequestControlContextImpl.java:214) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.TransitionableState.handleEvent(TransitionableState.java:116) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.Flow.handleEvent(Flow.java:547) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.impl.FlowExecutionImpl.handleEvent(FlowExecutionImpl.java:390) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.impl.RequestControlContextImpl.handleEvent(RequestControlContextImpl.java:210) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.ActionState.doEnter(ActionState.java:105) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.State.enter(State.java:194) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.Transition.execute(Transition.java:228) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.impl.FlowExecutionImpl.execute(FlowExecutionImpl.java:395) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.impl.RequestControlContextImpl.execute(RequestControlContextImpl.java:214) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.TransitionableState.handleEvent(TransitionableState.java:116) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.Flow.handleEvent(Flow.java:547) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.impl.FlowExecutionImpl.handleEvent(FlowExecutionImpl.java:390) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.impl.RequestControlContextImpl.handleEvent(RequestControlContextImpl.java:210) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.ActionState.doEnter(ActionState.java:105) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.State.enter(State.java:194) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.Transition.execute(Transition.java:228) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.impl.FlowExecutionImpl.execute(FlowExecutionImpl.java:395) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.impl.RequestControlContextImpl.execute(RequestControlContextImpl.java:214) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.TransitionableState.handleEvent(TransitionableState.java:116) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.Flow.handleEvent(Flow.java:547) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.impl.FlowExecutionImpl.handleEvent(FlowExecutionImpl.java:390) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.impl.RequestControlContextImpl.handleEvent(RequestControlContextImpl.java:210) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.ViewState.handleEvent(ViewState.java:231) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.ViewState.resume(ViewState.java:195) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.Flow.resume(Flow.java:537) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.engine.impl.FlowExecutionImpl.resume(FlowExecutionImpl.java:259) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.executor.FlowExecutorImpl.resumeExecution(FlowExecutorImpl.java:169) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.mvc.servlet.FlowHandlerAdapter.handle(FlowHandlerAdapter.java:228) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:959) ~[spring-webmvc-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:893) ~[spring-webmvc-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:970) ~[spring-webmvc-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:872) ~[spring-webmvc-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:648) ~[servlet-api.jar:?]
        at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:846) ~[spring-webmvc-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:729) ~[servlet-api.jar:?]
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:292) ~[catalina.jar:8.0.33]
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207) ~[catalina.jar:8.0.33]
        at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52) ~[tomcat-websocket.jar:8.0.33]
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240) ~[catalina.jar:8.0.33]
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207) ~[catalina.jar:8.0.33]
        at org.jasig.cas.security.ResponseHeadersEnforcementFilter.doFilter(ResponseHeadersEnforcementFilter.java:227) ~[cas-server-security-filter-2.0.4.jar:2.0.4]
        at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:346) ~[spring-web-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:262) ~[spring-web-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240) ~[catalina.jar:8.0.33]
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207) ~[catalina.jar:8.0.33]
        at org.jasig.cas.security.RequestParameterPolicyEnforcementFilter.doFilter(RequestParameterPolicyEnforcementFilter.java:250) ~[cas-server-security-filter-2.0.4.jar:2.0.4]
        at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:346) ~[spring-web-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:262) ~[spring-web-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240) ~[catalina.jar:8.0.33]
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207) ~[catalina.jar:8.0.33]
        at org.jasig.inspektr.common.web.ClientInfoThreadLocalFilter.doFilter(ClientInfoThreadLocalFilter.java:62) ~[inspektr-common-1.3.GA.jar:1.3.GA]
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240) ~[catalina.jar:8.0.33]
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207) ~[catalina.jar:8.0.33]
        at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:121) ~[spring-web-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107) ~[spring-web-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:346) ~[spring-web-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:262) ~[spring-web-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:240) ~[catalina.jar:8.0.33]
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:207) ~[catalina.jar:8.0.33]
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:212) ~[catalina.jar:8.0.33]
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:106) ~[catalina.jar:8.0.33]
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:502) ~[catalina.jar:8.0.33]
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:141) ~[catalina.jar:8.0.33]
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:79) ~[catalina.jar:8.0.33]
        at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:616) ~[catalina.jar:8.0.33]
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:88) ~[catalina.jar:8.0.33]
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:522) ~[catalina.jar:8.0.33]
        at org.apache.coyote.ajp.AbstractAjpProcessor.process(AbstractAjpProcessor.java:868) ~[tomcat-coyote.jar:8.0.33]
        at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:672) ~[tomcat-coyote.jar:8.0.33]
        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1502) ~[tomcat-coyote.jar:8.0.33]
        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.run(NioEndpoint.java:1458) ~[tomcat-coyote.jar:8.0.33]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) [?:1.7.0_101]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615) [?:1.7.0_101]
        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) ~[tomcat-util.jar:8.0.33]
        at java.lang.Thread.run(Thread.java:745) [?:1.7.0_101]
Caused by: java.lang.NullPointerException
        at org.jasig.cas.authentication.principal.DefaultPrincipalAttributesRepository.getPrincipalAttributes(DefaultPrincipalAttributesRepository.java:28) ~[cas-server-core-authentication-4.2.1.jar:4.2.1]
        at org.jasig.cas.authentication.principal.cache.AbstractPrincipalAttributesRepository.getAttributes_aroundBody2(AbstractPrincipalAttributesRepository.java:189) ~[cas-server-core-authentication-4.2.1.jar:4.2.1]
        at org.jasig.cas.authentication.principal.cache.AbstractPrincipalAttributesRepository$AjcClosure3.run(AbstractPrincipalAttributesRepository.java:1) ~[cas-server-core-authentication-4.2.1.jar:4.2.1]
        at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjweaver-1.8.6.jar:1.8.6]
        at org.jasig.inspektr.aspect.TraceLogAspect.traceMethod(TraceLogAspect.java:44) ~[inspektr-aspects-1.3.GA.jar:1.3.GA]
        at org.jasig.cas.authentication.principal.cache.AbstractPrincipalAttributesRepository.getAttributes(AbstractPrincipalAttributesRepository.java:188) ~[cas-server-core-authentication-4.2.1.jar:4.2.1]
        at org.jasig.cas.services.AbstractRegisteredServiceAttributeReleasePolicy.getAttributes_aroundBody8(AbstractRegisteredServiceAttributeReleasePolicy.java:82) ~[cas-server-core-authentication-4.2.1.jar:4.2.1]
        at org.jasig.cas.services.AbstractRegisteredServiceAttributeReleasePolicy$AjcClosure9.run(AbstractRegisteredServiceAttributeReleasePolicy.java:1) ~[cas-server-core-authentication-4.2.1.jar:4.2.1]
        at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjweaver-1.8.6.jar:1.8.6]
        at org.jasig.inspektr.aspect.TraceLogAspect.traceMethod(TraceLogAspect.java:44) ~[inspektr-aspects-1.3.GA.jar:1.3.GA]
        at org.jasig.cas.services.AbstractRegisteredServiceAttributeReleasePolicy.getAttributes(AbstractRegisteredServiceAttributeReleasePolicy.java:80) ~[cas-server-core-authentication-4.2.1.jar:4.2.1]
        at org.jasig.cas.CentralAuthenticationServiceImpl.grantServiceTicket_aroundBody2(CentralAuthenticationServiceImpl.java:164) ~[cas-server-core-4.2.1.jar:4.2.1]
        at org.jasig.cas.CentralAuthenticationServiceImpl$AjcClosure3.run(CentralAuthenticationServiceImpl.java:1) ~[cas-server-core-4.2.1.jar:4.2.1]
        at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjweaver-1.8.6.jar:1.8.6]
        at org.jasig.inspektr.aspect.TraceLogAspect.traceMethod(TraceLogAspect.java:44) ~[inspektr-aspects-1.3.GA.jar:1.3.GA]
        at org.jasig.cas.CentralAuthenticationServiceImpl.grantServiceTicket(CentralAuthenticationServiceImpl.java:136) ~[cas-server-core-4.2.1.jar:4.2.1]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.7.0_101]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57) ~[?:1.7.0_101]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.7.0_101]
        at java.lang.reflect.Method.invoke(Method.java:606) ~[?:1.7.0_101]
        at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:302) ~[spring-aop-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:190) ~[spring-aop-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:157) ~[spring-aop-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:85) ~[spring-aop-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at org.jasig.inspektr.audit.AuditTrailManagementAspect.handleAuditTrail(AuditTrailManagementAspect.java:128) ~[inspektr-audit-1.3.GA.jar:1.3.GA]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.7.0_101]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57) ~[?:1.7.0_101]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.7.0_101]
        at java.lang.reflect.Method.invoke(Method.java:606) ~[?:1.7.0_101]
        at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:621) ~[spring-aop-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:610) ~[spring-aop-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:68) ~[spring-aop-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:168) ~[spring-aop-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:99) ~[spring-tx-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:281) ~[spring-tx-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:96) ~[spring-tx-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179) ~[spring-aop-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:92) ~[spring-aop-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179) ~[spring-aop-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at com.ryantenney.metrics.spring.MeteredMethodInterceptor.invoke(MeteredMethodInterceptor.java:45) ~[metrics-spring-3.1.2.jar:?]
        at com.ryantenney.metrics.spring.MeteredMethodInterceptor.invoke(MeteredMethodInterceptor.java:32) ~[metrics-spring-3.1.2.jar:?]
        at com.ryantenney.metrics.spring.AbstractMetricMethodInterceptor.invoke(AbstractMetricMethodInterceptor.java:59) ~[metrics-spring-3.1.2.jar:?]
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179) ~[spring-aop-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at com.ryantenney.metrics.spring.TimedMethodInterceptor.invoke(TimedMethodInterceptor.java:48) ~[metrics-spring-3.1.2.jar:?]
        at com.ryantenney.metrics.spring.TimedMethodInterceptor.invoke(TimedMethodInterceptor.java:34) ~[metrics-spring-3.1.2.jar:?]
        at com.ryantenney.metrics.spring.AbstractMetricMethodInterceptor.invoke(AbstractMetricMethodInterceptor.java:59) ~[metrics-spring-3.1.2.jar:?]
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179) ~[spring-aop-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at com.ryantenney.metrics.spring.CountedMethodInterceptor.invoke(CountedMethodInterceptor.java:46) ~[metrics-spring-3.1.2.jar:?]
        at com.ryantenney.metrics.spring.CountedMethodInterceptor.invoke(CountedMethodInterceptor.java:32) ~[metrics-spring-3.1.2.jar:?]
        at com.ryantenney.metrics.spring.AbstractMetricMethodInterceptor.invoke(AbstractMetricMethodInterceptor.java:59) ~[metrics-spring-3.1.2.jar:?]
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179) ~[spring-aop-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:208) ~[spring-aop-4.2.3.RELEASE.jar:4.2.3.RELEASE]
        at com.sun.proxy.$Proxy77.grantServiceTicket(Unknown Source) ~[?:?]
        at org.jasig.cas.web.flow.GenerateServiceTicketAction.doExecute(GenerateServiceTicketAction.java:79) ~[cas-server-webapp-actions-4.2.1.jar:4.2.1]
        at org.springframework.webflow.action.AbstractAction.execute(AbstractAction.java:188) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        at org.springframework.webflow.execution.ActionExecutor.execute(ActionExecutor.java:51) ~[spring-webflow-2.4.2.RELEASE.jar:2.4.2.RELEASE]
        ... 87 more
`
Thank you.
Claude

--- a/cas-server-core-authentication/src/main/java/org/jasig/cas/services/AbstractRegisteredServiceAttributeReleasePolicy.java
+++ b/cas-server-core-authentication/src/main/java/org/jasig/cas/services/AbstractRegisteredServiceAttributeReleasePolicy.java
@@ -22,7 +22,6 @@ public abstract class AbstractRegisteredServiceAttributeReleasePolicy implements
     private static final long serialVersionUID = 5325460875620586503L;
 
     /** The LOGGER. */
- //   protected final transient Logger logger = LoggerFactory.getLogger(this.getClass());
     private static final Logger LOGGER = LoggerFactory.getLogger(AbstractRegisteredServiceAttributeReleasePolicy.class);
     /** The attribute filter. */
     private RegisteredServiceAttributeFilter registeredServiceAttributeFilter;

--- a/cas-server-core-authentication/src/main/java/org/jasig/cas/services/AbstractRegisteredServiceAttributeReleasePolicy.java
+++ b/cas-server-core-authentication/src/main/java/org/jasig/cas/services/AbstractRegisteredServiceAttributeReleasePolicy.java
@@ -12,7 +12,7 @@ import org.slf4j.LoggerFactory;
 import java.util.Map;
 
 /**
- * Abstract release policy for attributes, provides common shared settings such as loggers and attribute filter config.
+ * Abstract release policy for attributes, provides common shared settings such as LOGGERs and attribute filter config.
  * Subclasses are to provide the behavior for attribute retrieval.
  * @author Misagh Moayyed
  * @since 4.1.0
@@ -21,9 +21,9 @@ public abstract class AbstractRegisteredServiceAttributeReleasePolicy implements
     
     private static final long serialVersionUID = 5325460875620586503L;
 
-    /** The logger. */
-    protected final transient Logger logger = LoggerFactory.getLogger(this.getClass());
-
+    /** The LOGGER. */
+ //   protected final transient Logger logger = LoggerFactory.getLogger(this.getClass());
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractRegisteredServiceAttributeReleasePolicy.class);
     /** The attribute filter. */
     private RegisteredServiceAttributeFilter registeredServiceAttributeFilter;
 

--- a/cas-server-core-authentication/src/main/java/org/jasig/cas/services/AbstractRegisteredServiceAttributeReleasePolicy.java
+++ b/cas-server-core-authentication/src/main/java/org/jasig/cas/services/AbstractRegisteredServiceAttributeReleasePolicy.java
@@ -12,7 +12,7 @@ import org.slf4j.LoggerFactory;
 import java.util.Map;
 
 /**
- * Abstract release policy for attributes, provides common shared settings such as LOGGERs and attribute filter config.
+ * Abstract release policy for attributes, provides common shared settings such as loggers and attribute filter config.
  * Subclasses are to provide the behavior for attribute retrieval.
  * @author Misagh Moayyed
  * @since 4.1.0

--- a/cas-server-core-authentication/src/main/java/org/jasig/cas/services/ReturnAllowedAttributeReleasePolicy.java
+++ b/cas-server-core-authentication/src/main/java/org/jasig/cas/services/ReturnAllowedAttributeReleasePolicy.java
@@ -3,6 +3,8 @@ package org.jasig.cas.services;
 import org.apache.commons.lang3.builder.EqualsBuilder;
 import org.apache.commons.lang3.builder.HashCodeBuilder;
 import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.util.ArrayList;
 import java.util.Collections;
@@ -19,7 +21,9 @@ import java.util.Map;
 public final class ReturnAllowedAttributeReleasePolicy extends AbstractRegisteredServiceAttributeReleasePolicy {
 
     private static final long serialVersionUID = -5771481877391140569L;
-    
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ReturnAllowedAttributeReleasePolicy.class);
+
     private List<String> allowedAttributes;
 
     /**
@@ -64,7 +68,7 @@ public final class ReturnAllowedAttributeReleasePolicy extends AbstractRegistere
             final Object value = resolvedAttributes.get(attribute);
 
             if (value != null) {
-                logger.debug("Found attribute [{}] in the list of allowed attributes", attribute);
+                LOGGER.debug("Found attribute [{}] in the list of allowed attributes", attribute);
                 attributesToRelease.put(attribute, value);
             }
         }

--- a/cas-server-core-authentication/src/main/java/org/jasig/cas/services/ReturnMappedAttributeReleasePolicy.java
+++ b/cas-server-core-authentication/src/main/java/org/jasig/cas/services/ReturnMappedAttributeReleasePolicy.java
@@ -3,6 +3,8 @@ package org.jasig.cas.services;
 import org.apache.commons.lang3.builder.EqualsBuilder;
 import org.apache.commons.lang3.builder.HashCodeBuilder;
 import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.util.HashMap;
 import java.util.Map;
@@ -18,6 +20,8 @@ public class ReturnMappedAttributeReleasePolicy extends AbstractRegisteredServic
 
     private static final long serialVersionUID = -6249488544306639050L;
     
+    private static final Logger LOGGER = LoggerFactory.getLogger(ReturnMappedAttributeReleasePolicy.class);
+
     private Map<String, String> allowedAttributes;
 
     /**
@@ -64,7 +68,7 @@ public class ReturnMappedAttributeReleasePolicy extends AbstractRegisteredServic
 
             if (value != null) {
                 final String mappedAttributeName = entry.getValue();
-                logger.debug("Found attribute [{}] in the list of allowed attributes, mapped to the name [{}]",
+                LOGGER.debug("Found attribute [{}] in the list of allowed attributes, mapped to the name [{}]",
                         key, mappedAttributeName);
                 attributesToRelease.put(mappedAttributeName, value);
             }

--- a/cas-server-core-services/src/main/java/org/jasig/cas/services/DefaultRegisteredServiceUsernameProvider.java
+++ b/cas-server-core-services/src/main/java/org/jasig/cas/services/DefaultRegisteredServiceUsernameProvider.java
@@ -13,7 +13,6 @@ import org.slf4j.LoggerFactory;
  */
 public final class DefaultRegisteredServiceUsernameProvider implements RegisteredServiceUsernameAttributeProvider {
     private static final long serialVersionUID = 5823989148794052951L;
-//    private final transient Logger logger = LoggerFactory.getLogger(this.getClass());
     private static final Logger LOGGER = LoggerFactory.getLogger(DefaultRegisteredServiceUsernameProvider.class);
 
     @Override

--- a/cas-server-core-services/src/main/java/org/jasig/cas/services/DefaultRegisteredServiceUsernameProvider.java
+++ b/cas-server-core-services/src/main/java/org/jasig/cas/services/DefaultRegisteredServiceUsernameProvider.java
@@ -13,11 +13,12 @@ import org.slf4j.LoggerFactory;
  */
 public final class DefaultRegisteredServiceUsernameProvider implements RegisteredServiceUsernameAttributeProvider {
     private static final long serialVersionUID = 5823989148794052951L;
-    private final transient Logger logger = LoggerFactory.getLogger(this.getClass());
+//    private final transient Logger logger = LoggerFactory.getLogger(this.getClass());
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultRegisteredServiceUsernameProvider.class);
 
     @Override
     public String resolveUsername(final Principal principal, final Service service) {
-        logger.debug("Returning the default principal id [{}] for username.", principal.getId());
+        LOGGER.debug("Returning the default principal id [{}] for username.", principal.getId());
         return principal.getId();
     }
 

